### ISSUE TITLE
Can't install/update extensions in Brackets; Installation failed "Unknown internal error"

### ISSUE THREAD

**Issue by joauv** [adobe/brackets#10266]
I've been having problems with installing/updating extensions for a while:
![image](https://cloud.githubusercontent.com/assets/7191186/5531501/84b43234-89fa-11e4-920b-342b318d9268.png)

I followed the steps to troubleshoot the problem, and I eventually ended up deleting all of my extensions, clearing the cache, deleting and re-installing Brackets... I thought Node.js might be a problem at some point, and also read up on proxy stuff, but now I'm clueless as to what causes this "unknown internal error" when I try to install an extension. Even manually installing extensions doesn't work now.

I've already tried every "solution" I could find, and nothing works. So... yeah, it's getting really frustrating.

**Comment by busykai** [comment]
@joauv, it sounds like `localhost` cannot be resolved on your machine. Can you do `dig localhost` or `ping localhost` successfully? Does it return `127.0.0.1` (or resolve in case of `ping`) as the address for `localhost`? If not, then you could try follow the solution steps [in this thread](https://discussions.apple.com/thread/2729411?tstart=0) to recover.

**Comment by joauv** [comment]
That was it @busykai .

For some reason, I had no "hosts" file in my /etc/ folder... I simply created one and placed it there... and magic, everything works again.

**Comment by busykai** [comment]
@joauv, glad it helped. I think we can safely close this issue.

**Comment by redmunds** [comment]
@busykai Is this worth mentioning in the [Troubleshooting guide](https://github.com/adobe/brackets/wiki/Troubleshooting#extension-install)?

**Comment by busykai** [comment]
@redmunds, I was thinking about it and couldn't decide. Let's put it there. I'll take care of it.

**Comment by MarcelGerber** [comment]
@redmunds @busykai I wonder if we should just change `localhost` to `127.0.0.1` in `NodeConnection.js` (there is another occurrence in MultiBrowserLiveDev, but that's less important for now).
That'd probably have avoided this issue, and hopefully, it can also fix some of the other extension installation/live preview issues (we had quite a lot of them recently).

Is that okay with you?

**Comment by busykai** [comment]
@MarcelGerber, `localhost` is a general convention for a loop-back networking interface, it's actual IP is not guaranteed to be `127.0.0.1`. *However*, we do listen to `127.0.0.1` (instead of `localhost`) -- `StaticServer` actually uses `127.0.0.1` literal. That part should be changed instead, I believe.

**Comment by MarcelGerber** [comment]
Hm, but in case it can prevent issues, imho it's worth changing `localhost` to 127.0.0.1.
We could also introduce a new pref for the few case where localhost isn't 127.0.0.1

**Comment by busykai** [comment]
My logic is people who use a different IP address (any of `127.0.0.0/8`) for `localhost` and their hostname resolution correctly configured should be able to use Brackets, we should not tell them to have `127.0.0.1` for their loopback interfaces. For the cases where host resolution is broken, such as this issue, we should be able to diagnose it better.

I don't think that switching to literal IP is a good solution.

**Comment by aroberge** [comment]
I have been getting the same error message.  In my case, uninstalling an anti-virus program made all the difference: http://aroberge.blogspot.ca/2015/01/disgusted-by-lavasoft-adaware-antivirus.html

**Comment by AlifArnado** [comment]
I have this problem also...
help me please

**Comment by nethip** [comment]
@AlifArnado Did you try steps mentioned by @aroberge ?

**Comment by AlifArnado** [comment]
@nethip I tried on a 64 -bit Linux OS elementary ,, what steps it enables successful I have tried it and the results are not there ,, please help

**Comment by jbeckton** [comment]
just fired up brackets and updated it, I rarely use it. I was able to install a couple extensions and then i started having this issue. Reminds me why I don't use it much.

**Comment by nethip** [comment]
@jbeckton Did you try troubleshooting steps mentioned in the earlier comments on this thread. You could check for any anti virus program running on your system that might be blocking extension manager. Also check `/etc/hosts` file if you see something fishy over there. Also please post the console log that you see in the developer tools.

**Comment by thoroc** [comment]
Thanks I was able to resolve my issue with Brackets under Windows 7 x64: Basically Brackets could not connect to the WebSocket on localhost. This was being blocked by the AV software. I added a rule for this connection and Brackets is now happy. And so am I. ^_^

**Comment by ForeverPx** [comment]
I fix the problem by delete the brackets folder in /Library/Application Support/Brackets

**Comment by AlifArnado** [comment]
thank's for all problem solve is done

**Comment by Narayanak** [comment]
Uninstall chrome and Brackets and reinstall ...It works for me

**Comment by AlifArnado** [comment]
no you can try install node.js

**Comment by O-Ccdream** [comment]
me too Uninstall all of kind

**Comment by O-Ccdream** [comment]
另外，哪个国人看懂了给翻译下 英语学的不好啊

**Comment by tv-nieri** [comment]
I hope this helps someone...

I had this issue also. In my case, i got the .deb package from brackts.io website, the x64 version.
You can install .deb packages with the Software Center in Ubuntu, but i used the dpkg command insted, like this exemple:
$sudo dpkg -i package.deb

When i tried to install some extensions, i found this issue, so i was going to try place the .zip packages in the extension folder, under Help > Extension Folder. The OS requested my password, that's when i figured out, the dir under ~/.config/Brackets has the root as owner,so i chaged it:
$cd ~/.config
$sudo chown -R user:group Brackets

This solved my problem. You can also maintain the root as owner of this dir, and change the permissions, i prefer to change the ownership from root to my user.

**Comment by muhabmaselim** [comment]
I have the same problem. I am running brackets 1.8 on Windows 10 32-bit from a private network and this is what I have done so far:
1.Made sure the 127.0.0.1 is defined as localhost in C:]Windows]System32\Drivers]etc]hosts
2.Turned off my Windows Firewall
3.Made sure brackets is allowed when using a private network through Control Panel\System and Security\Allow an app through Windows Firewall
4.Made sure Brackets.exe has all inbound and outbound rules enabled on a private network via Control Panel\System and Security\Windows Firewall\Advanced Settings
5.Uninstalled and reinstalled Brackets
6.Ran Brackets.exe as administrator
7.Dragged and dropped .zip file to the drag and drop box in the extension manager
8.Provided Github URL for the JSON Lint Extension for Brackets (https://github.com/ingorichter/de.richter.brackets.jsonlint) to the install from URL section
9.Unzipped the JSON Lint extension zip file and copied and pasted its contents to Brackets extension folder

I am still getting 'Unknown Error'. What am I missing?

**Comment by zaggino** [comment]
@muhabmaselim see last comments here https://github.com/adobe/brackets/issues/12898

**Comment by muhabmaselim** [comment]
@zaggino It was the Node.js! It is working now, finally! I must have spent hours trying to get it to work. Thank you for linking me to that thread.

**Comment by ficristo** [comment]
Original issue was about something else. Locking to avoid further comments on a closed fixed issue.
Feel free to open a new issue.
